### PR TITLE
check that roomstate doesn't keep stale connections

### LINKED ISSUE #46: Make tunnel client testing more comprehensive [Rooms 1.0 Completion]
Right now `tests/nodejs` just verifies that a connection between two peers can be established.

We should also test

* [x] peers not on the members list can't connect (implemented but not tested)
* [x] what happens with peers that can't be dialed by the room server (tunnel.connect fails or times out)
* [x] no staleness in roomstate
  * [x] disconnect and be removed from the current connected peers
  * [x] members disconnect and connecting again, make new tunnels

### PR DESCRIPTION
**Pull request by cryptix**
fixes #46

### CODE REVIEW

File: muxrpc/test/go/roomstate_test.go [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+	t.Cleanup(cancel)
+
+	// create the roomsrv
+	ts := makeNamedTestBot(t, "server", ctx, nil)
+	ctx = ts.ctx
+
+	// two new clients, connected to server automaticaly
+	// default to member during makeNamedTestBot
+	tal := ts.makeTestClient("tal")
+	srh := ts.makeTestClient("srh")
```
Comment (cblgh) [inline muxrpc/test/go/roomstate_test.go:38]: wtf is srh and tal

File: muxrpc/test/go/roomstate_test.go [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+	t.Log("restarted srh")
+
+	time.Sleep(1 * time.Second) // let server respond
+
+	// announce srh so that tal can connect
+	err = srh.Async(ctx, &ok, muxrpc.TypeJSON, muxrpc.Method{"room", "announce"})
+	r.NoError(err)
+	r.True(ok)
+
+	testKexReply := []byte("fake kex reply")
```
Comment (cblgh) [inline muxrpc/test/go/roomstate_test.go:87]: image

File: muxrpc/test/go/roomstate_test.go (hunk 1 above)
Comment (cryptix) [reply muxrpc/test/go/roomstate_test.go:38]: oh shit. it's just names. i was bored with alice and bob.

Comment (cblgh) [reply muxrpc/test/go/roomstate_test.go:38]: haha oh okay ^^

**Review by staltz — APPROVED** [review]

File: muxrpc/test/go/roomstate_test.go (hunk 1 above)
Comment (cblgh) [reply muxrpc/test/go/roomstate_test.go:38]: alice and bob are out, tal and srh are IN

Comment (cryptix) [reply muxrpc/test/go/roomstate_test.go:38]: https://en.wikipedia.org/wiki/Sarah
https://en.wikipedia.org/wiki/Tal_(name)

File: muxrpc/test/go/roomstate_test.go [hunk 3]
```diff
@@ -31,13 +34,14 @@ func TestStaleMembers(t *testing.T) {
 
 	// two new clients, connected to server automaticaly
 	// default to member during makeNamedTestBot
-	tal := ts.makeTestClient("tal")
-	srh := ts.makeTestClient("srh")
+	tal := ts.makeTestClient("tal") // https://en.wikipedia.org/wiki/Tal_(name)
+	srh := ts.makeTestClient("srh") // https://en.wikipedia.org/wiki/Sarah
```
Comment (cblgh) [inline muxrpc/test/go/roomstate_test.go:38]: 💯

**Review by cblgh — APPROVED** [review]
lgtm, i don't really have any major comments :)

**Comment by cblgh** [comment]
🔥 🔥

**Pull request merged**